mp/src/game/client/momentum/ui/HUD/hud_stickybombs.cpp

```diff
@@ -0,0 +1,119 @@
+#include "cbase.h"
+#include <vgui/ILocalize.h>
+#include <vgui/ISurface.h>
+#include <vgui/IVGui.h>
+#include <vgui_controls/EditablePanel.h>
+#include <vgui_controls/Label.h>
+#include "c_mom_player.h"
+#include "hud.h"
+#include "hudelement.h"
+#include "iclientmode.h"
+#include "ienginevgui.h"
+#include "mom_system_gamemode.h"
+#include "weapon/weapon_base.h"
+#include "weapon/weapon_mom_stickybomblauncher.h"
+
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+static MAKE_TOGGLE_CONVAR(mom_sj_stickycount_enable, "1", FCVAR_ARCHIVE, "Toggles the stickybomb counter.\n");
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+class CHudStickybombs : public CHudElement, public EditablePanel
+{
+    DECLARE_CLASS_SIMPLE(CHudStickybombs, EditablePanel);
+
+  public:
+    CHudStickybombs(const char *pElementName);
+
+    void ApplySchemeSettings(IScheme *scheme) OVERRIDE;
+    bool ShouldDraw() OVERRIDE;
+    void OnTick() OVERRIDE;
+
+  private:
+    vgui::EditablePanel *m_pStickybombsPresent;
+    vgui::EditablePanel *m_pNoStickybombsPresent;
+    vgui::Label *m_pStickyLabel;
+    vgui::Label *m_pNoStickyLabel;
+};
+
+DECLARE_HUDELEMENT(CHudStickybombs);
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+CHudStickybombs::CHudStickybombs(const char *pElementName)
+    : CHudElement(pElementName), BaseClass(NULL, "CHudStickybombs")
+{
+    Panel *pParent = g_pClientMode->GetViewport();
+    SetParent(pParent);
+
+    m_pStickybombsPresent = new EditablePanel(this, "StickybombsPresentPanel");
+    m_pNoStickybombsPresent = new EditablePanel(this, "NoStickybombsPresentPanel");
+    m_pStickyLabel = new Label(m_pStickybombsPresent, "NumStickybombsLabel", "");
+    m_pNoStickyLabel = new Label(m_pNoStickybombsPresent, "NumNoStickybombsLabel", "");
```

Comment:
Not sure what's going on here. Why the need for a completely different EditablePanel when there aren't any stickies? There's no need for an editable panel child either; this element is an editable panel.

Let's reduce this down to one Label, and use CHudStickyBombs as what the `m_pStickybombsPresent` is used for.

Reply:
Looking at the default TF2 HUD, it seems that the sticky icon on there gets dimmed when no stickies are out, which would explain the double panels, but since we probably won't have an icon on our HUD I will just cut it down to the one label.

Reply:
Yeah... plus we can always use panel alpha if we want something like that, a completely new panel UI element for such a simple effect is bad.